### PR TITLE
add pgh_operation field and delete snapshot event

### PR DESCRIPTION
**Pull request by nossila**
First, thanks a lot for this package, I really wanted to build something similar for quite sometime.

I have a requirement for knowing what was the operation (insert, update or delete) and also track deletes snapshot.

I ended up splitting `Snapshot` into 3 other classes `SnapshotInsert`, `SnapshotUpdate` and `SnapshotDelete` so we could also use just the ones you need, like: `@pghistory.track(pghistory.SnapshotDelete())`.

What you guys think about it? Let me know if this is an interesting contribution and if I can improve the code in anyway.

### CODE REVIEW

**Comment by wesleykendall** [comment]
@nossila thanks so much for your suggestion and code.

I've been unavailable for some time to contribute and am finally coming back around to merging in some of these PRs.

At a high level, I'm very cautious of accepting a change that modifies the default tracking tables. Some customers have hundreds of millions of snapshot rows, and they'd have to migrate all of them.

Let me come back to you with some thoughts on how we can try to handle this use case though. There may be something I didn't fully document that may support your use case well. 

One question that comes to mind though - Is it possible for you to determine this information dynamically? Correct me if I'm wrong, but the the first snapshot of a row is always going to be an INSERT,  everything in between will be UPDATE, and the final one will be DELETE if the original record no longer exists. I believe the only time this case doesn't apply is for legacy data where you weren't tracking events.

Let me know though. I will come back with some more thoughts

**Comment by wesleykendall** [comment]
Thought about it some more. I think this would fit better as a custom base model you set for every snapshot model in your project [using this setting](https://django-pghistory.readthedocs.io/en/2.5.1/settings.html#pghistory-base-model)

I could see this being added to the library though because this base model would be special. I.e. it would either need to install different tracking triggers or the current triggers would need to be aware of it. The tracking triggers could also be updated to look for a special `pgh_operation` field like you have.

In other words, the change wouldn't affect any current users, and we could put a section in the docs on this other base model for users that need this additional metadata.

Let me know if that makes sense and perhaps I can propose or make some changes to the PR.

**Comment by wesleykendall** [comment]
Hey wanted to follow up on this. `pghistory.track` is more configurable now, which I believe addresses the original intentions of this PR.

By default, `pghistory.track` will register  `pghistory.InsertEvent()` and `pghistory.UpdateEvent()` trackers. They both set `pgh_label` to `insert` or `update`, in effect accomplishing the functionality of tracking the type of operation without introducing a new field.

You can also override the trackers per-model (see [the docs here](https://django-pghistory.readthedocs.io/en/3.3.0/event_tracking/#custom-trackers)) or use the `PGHISTORY_DEFAULT_TRACKERS` setting to override the default trackers everywhere.

I cleaned up some of the config/docs a bit more in https://github.com/Opus10/django-pghistory/pull/142, which will be deployed shortly

Hope this helps!
